src/EditorFeatures/Core/Implementation/ExtractMethod/AbstractExtractMethodCommandHandler.cs

```diff
@@ -72,27 +72,28 @@ public VSCommanding.CommandState GetCommandState(ExtractMethodCommandArgs args)
 
         public bool ExecuteCommand(ExtractMethodCommandArgs args, CommandExecutionContext context)
         {
-            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
-            if (document == null)
+            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Applying_Extract_Method_refactoring))
             {
-                return false;
-            }
+                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoaddedOpenDocumentInCurrentContextWithChangesAsync(
+                    context.OperationContext).WaitAndGetResult(context.OperationContext.UserCancellationToken);
+                if (document == null)
+                {
+                    return false;
+                }
 
-            var supportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
-            if (!supportsFeatureService.SupportsRefactorings(document))
-            {
-                return false;
-            }
+                var supportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
+                if (!supportsFeatureService.SupportsRefactorings(document))
+                {
+                    return false;
+                }
 
-            // Finish any rename that had been started. We'll do this here before we enter the
-            // wait indicator for Extract Method
-            if (_renameService.ActiveSession != null)
-            {
-                _renameService.ActiveSession.Commit();
-            }
+                // Finish any rename that had been started. We'll do this here before we enter the
+                // wait indicator for Extract Method
+                if (_renameService.ActiveSession != null)
+                {
+                    _renameService.ActiveSession.Commit();
+                }
 
-            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Applying_Extract_Method_refactoring))
-            {
                 return Execute(args.SubjectBuffer, args.TextView, context.OperationContext);
```

Comment:
probably best to pass in Document here.  It's not really a great approach to have multple documents floating around that may or may not represent hte same info.

Reply:
I looked all command handler. some reason all do this pattern. until I know why they are doing that pattern, I will leave it as they are. and changing all those command handler will be another refactoring we can do isolated from this change.